### PR TITLE
#1526 Add Twilio and ASPSMS SMS providers support + new notification type

### PR DESCRIPTION
**Pull request by yecli**
Added SMS providers and new notification template

### CODE REVIEW

File: VirtoCommerce.Platform.Web/Startup.cs [hunk 1]
```diff
@@ -529,6 +541,62 @@ private static void InitializePlatform(IAppBuilder app, IUnityContainer containe
                         }
                     },
                     new ModuleSettingsGroup
+                    {
```
Comment (tatarincev) [inline VirtoCommerce.Platform.Web/Startup.cs]: We don't need a create platform settings that can be changed in run-time, let's restricting only settings that being  reading from web.config

Comment (yecli) [reply VirtoCommerce.Platform.Web/Startup.cs]: Done

File: VirtoCommerce.Platform.Data.Notifications/SMSGatewayOptions.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+namespace VirtoCommerce.Platform.Data.Notifications
+{
+    public class SmsGatewayOptions
```
Comment (tatarincev) [inline VirtoCommerce.Platform.Data.Notifications/SMSGatewayOptions.cs]: I think that make sense create the two different Options for each provider type and inject or pass into ctor of concrete provider only  settings that it understand.
Factory for this options you can implement in startup as well

Comment (yecli) [reply VirtoCommerce.Platform.Data.Notifications/SMSGatewayOptions.cs]: Committed changes regarding this. If it is too complex - could add two different classes and option sets.

**Pull request merged**